src/main/java/ManageInputs.java

```diff
@@ -0,0 +1,209 @@
+            String description = line.substring(6, indexFrom - 1);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("event description not specified");
+            throw new UnexpectedCommandException();
+        }
+        String from = line.substring(indexFrom + 5, indexTo - 1);
+        String to = line.substring(indexTo + 2);
+        String description = line.substring(5, indexFrom - 1);
+
+        tasks.add(index, new Event(description, from, to));
+    }
+
+    public static void dealWithDeadline(ArrayList<Task> tasks, int index, String line) throws UnexpectedCommandException, IOException {
+        int indexBy = line.indexOf("by");
+        int space = line.indexOf(" ");
+        if (indexBy == -1) {//invalid format
+            System.out.println("Invalid format! Enter deadline in the format: deadline (description) by (deadline)");
+            throw new UnexpectedCommandException();
+        }
+        try {//deadline / both not specified
+            String by = line.substring(indexBy + 2);
+        } catch (IndexOutOfBoundsException e) {
+            try {
+                String description = line.substring(space, indexBy - 1);
+            } catch (IndexOutOfBoundsException f) {
+                System.out.println("deadline description and deadline not specified");
+                throw new UnexpectedCommandException();
+            }
+            System.out.println("deadline not specified");
+            throw new UnexpectedCommandException();
+        }
+        try {//deadline not specified
+            String description = line.substring(space, indexBy - 1);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("deadline description not specified");
+            throw new UnexpectedCommandException();
+        }
+        String description = line.substring(space, indexBy - 1);
+        String by = line.substring(indexBy + 2);
+
+        tasks.add(index, new Deadline(description, by));
+    }
+
+    public static void dealWithTodo(ArrayList<Task> tasks, int index, String line) throws UnexpectedCommandException, IOException {
+        int indexSpace = line.indexOf(" ");
+        if (indexSpace == -1) {
+            System.out.println("todo description not specified");
+            throw new UnexpectedCommandException();
+        }
+
+        String description = line.substring(indexSpace);
+
+        tasks.add(index, new Todo(description));
+
+    }
+
+    private void handleUnexpectedCommand(boolean isValidCommand) throws UnexpectedCommandException {
+        if (!isValidCommand) {
+            throw new UnexpectedCommandException();
+        }
+    }
+
+    private void handleEmptyInput(String line) throws EmptyLineException {
+        if (line.isEmpty()) {
+            throw new EmptyLineException();
+        }
+    }
+
+    public ManageInputs(ArrayList<Task> tasks, int index, String line) throws IOException, UnexpectedCommandException {
```

Comment:
This function is doing quite a fair bit of heavy lifting.  

It's very long, which usually indicates that perhaps you could apply [SLAP](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) more here. 

To aid you in achieving SLAP, here are some guiding questions...
1. What parts of the code is repetitive?  i.e. virtually the same for each if/else block you have
2. What part of the code does a particular function that can be isolated into a function? (like printing tasks, or doing something to the file?)